test/Concurrency/transfernonsendable_strong_transferring.swift

```diff
@@ -0,0 +1,228 @@
+// RUN: %target-swift-frontend -emit-sil -disable-availability-checking -enable-experimental-feature TransferringArgsAndResults -verify -enable-experimental-feature RegionBasedIsolation %s
+
+// REQUIRES: asserts
+
+////////////////////////
+// MARK: Declarations //
+////////////////////////
+
+class Klass {}
+
+struct NonSendableStruct {
+  var first = Klass()
+  var second = Klass()
+}
+
+func useValue<T>(_ t: T) {}
+func getAny() -> Any { fatalError() }
+@MainActor func transferToMain<T>(_ t: T) {}
+
+func transferArg(_ x: transferring Klass) {
+}
+
+func transferArgWithOtherParam(_ x: transferring Klass, _ y: Klass) {
+}
+
+func transferArgWithOtherParam2(_ x: Klass, _ y: transferring Klass) {
+}
+
+@MainActor var globalKlass = Klass()
+
+/////////////////
+// MARK: Tests //
+/////////////////
+
+func testSimpleTransfer() {
+  let k = Klass()
+  transferArg(k) // expected-warning {{passing argument of non-sendable type 'Klass' from nonisolated context to nonisolated context at this call site could yield a race with accesses later in this function}}
+  useValue(k) // expected-note {{access here could race}}
+}
+
+func testSimpleTransferUseOfOtherParamNoError() {
+  let k = Klass()
+  let k2 = Klass()
+  transferArgWithOtherParam(k, k2)
+  useValue(k2)
+}
+
+func testSimpleTransferUseOfOtherParamNoError2() {
+  let k = Klass()
+  let k2 = Klass()
+  transferArgWithOtherParam2(k, k2)
+  useValue(k)
+}
+
+@MainActor func transferToMain2(_ x: transferring Klass, _ y: Klass, _ z: Klass) async {
+
+}
+
+// TODO: How to test this?
+func testNonStrongTransferDoesntMerge() async {
+}
+
+//////////////////////////////////
+// MARK: Transferring Parameter //
+//////////////////////////////////
+
+func testTransferringParameter_canTransfer(_ x: transferring Klass, _ y: Klass) async {
+  await transferToMain(x)
+  await transferToMain(y) // expected-warning {{call site passes `self` or a non-sendable argument of this function to another thread, potentially yielding a race with the caller}}
```

Comment:
I'm wondering if the warning should be phrased in terms of actor boundary crossing and not "to another thread", or is this intentional?

Reply:
I agree that said diagnostic needs to be wordsmithed (and actually split up into 1-2 other ones), this PR is just bringing up transferring, not solving that problem.

Reply:
Yea not a blocker for merging ofc, just something I noticed

Reply:
Thanks for bringing it up = ).